### ISSUE TITLE
Kubernetes test cases

### ISSUE THREAD

**Issue by rgmz** [leaktk/fake-leaks#3]
Are there any Kubernetes secrets outside of https://github.com/leaktk/fake-leaks/tree/main/k8s?

If not, I can contribute some more true/false positives.

**Comment by bplaxco** [comment]
It [looks](https://github.com/leaktk/patterns/blob/main/testdata/leaktk-scanner-results.yaml#L850-L860) like there's a few in general-test, but they should be moved under k8s too. 

I'm slowly working on restructuring that repo to be more organized. Most of it was dumped from automatic output from the old test framework that generated a lot of them on the fly.

Any interest in adding things and/or organizing is totally welcome!

Let me know if you have any questions about stuff along the way.

**Comment by bplaxco** [comment]
Also that link I sent above was from the patterns repo test cases, so they aren't used to generate any of the secrets, just what the scanner is picking up based off of it's patterns.

**Comment by rgmz** [comment]
Makes sense. I'll take a look and try to add anything I feel is missing.

So far I'm happy with https://github.com/gitleaks/gitleaks/pull/1462.

**Comment by bplaxco** [comment]
Ack, looking for me to pick through it? I might have a sec next week.

Aside: I moved my [script for generating regex for finding strings](https://github.com/leaktk/hack/blob/main/gen-base64-search-regex) in base64 over to my hack repo. It's really handy for finding JWTs with certain fields and such. That's mainly what I've been using it for.

**Comment by rgmz** [comment]
> Ack, looking for me to pick through it? I might have a sec next week.

Sure, if you have time! In general, it's nice to have a 'real' true/false positives outside of unit tests.

> Aside: I moved my [script for generating regex for finding strings](https://github.com/leaktk/hack/blob/main/gen-base64-search-regex) in base64 over to my hack repo. It's really handy for finding JWTs with certain fields and such. That's mainly what I've been using it for.

Ha, I [created a rule specifically for JWTs](https://github.com/gitleaks/gitleaks/blob/128cd22f7c7c8aaf001234999e509e84ef5d6577/cmd/generate/config/rules/jwt.go#L68). I wonder how well it performs compared to something like this.

**Comment by bplaxco** [comment]
I see you only have one chunk of b64 for each field. How was that generated? Just curious.

Like to capture `"alg":` I'd expect to see one of these three match in the wild depending on how the base64 is shifted:

```
(ImFsZyI6|JhbGciO|iYWxnIj)
```

(trying to see if there's a way to shorten my b64 matches.

**Comment by rgmz** [comment]
> I see you only have one chunk of b64 for each field. How was that generated? Just curious.

The horrifying truth is that I wrote that entire pattern by hand. [I didn't remember that I could generate it until I started making test cases.](https://github.com/gitleaks/gitleaks/pull/1256#issue-1865458113)

> ...depending on how the base64 is shifted:

My approach was to create permutations for `{` + each known key (alg, apu, exp, etc.). My assumption was that these short prefixes would be relatively stable, and anything that looked like prefix + `[\w+/]={0,3}` (base64 characters) would be a high confidence detection.

It's possible that my assumptions were incorrect and missed a bunch of cases.
